google/resource_resourcemanager_lien.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	// Store the ID now
+	id, err := replaceVars(d, config, "{{name}}")
+	if err != nil {
+		return fmt.Errorf("Error constructing id: %s", err)
+	}
+	d.SetId(id)
+
+	log.Printf("[DEBUG] Finished creating Lien %q: %#v", d.Id(), res)
+
+	d.SetId(res["name"].(string))
+	// Necessary so that we can match the Lien in the following Read().
+	d.Set("name", res["name"].(string))
+
+	return resourceResourceManagerLienRead(d, meta)
+}
+
+func resourceResourceManagerLienRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	url, err := replaceVars(d, config, "https://cloudresourcemanager.googleapis.com/v1/liens?parent={{parent}}")
+	if err != nil {
+		return err
+	}
+
+	res, err := Get(config, url)
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("ResourceManagerLien %q", d.Id()))
+	}
+
+	// Extract the object we're interested in from the list response.
+	itemsList_ := res["liens"]
+	var itemsList []interface{}
+	if itemsList_ != nil {
+		itemsList = itemsList_.([]interface{})
+	}
+	listObj := make([]map[string]interface{}, len(itemsList))
+	for i, item := range itemsList {
+		listObj[i] = item.(map[string]interface{})
+	}
+	res = nil
+	for _, item := range listObj {
+		thisName := d.Get("name")
+		thatName := flattenResourceManagerLienName(item["name"])
+		log.Printf("[DEBUG] Checking equality of %#v, %#v", thatName, thisName)
+		if !reflect.DeepEqual(thatName, thisName) {
+			continue
+		}
+		res = item
+		break
+	}
+	if res == nil {
+		// Object isn't there any more - remove it from the state.
+		log.Printf("[DEBUG] Removing ResourceManagerLien because it couldn't be matched.")
+		d.SetId("")
+		return nil
+	}
+
+	if err := d.Set("name", flattenResourceManagerLienName(res["name"])); err != nil {
+		return fmt.Errorf("Error reading Lien: %s", err)
+	}
+	if err := d.Set("reason", flattenResourceManagerLienReason(res["reason"])); err != nil {
+		return fmt.Errorf("Error reading Lien: %s", err)
+	}
+	if err := d.Set("origin", flattenResourceManagerLienOrigin(res["origin"])); err != nil {
+		return fmt.Errorf("Error reading Lien: %s", err)
+	}
+	if err := d.Set("create_time", flattenResourceManagerLienCreateTime(res["createTime"])); err != nil {
+		return fmt.Errorf("Error reading Lien: %s", err)
+	}
+	if err := d.Set("parent", flattenResourceManagerLienParent(res["parent"])); err != nil {
+		return fmt.Errorf("Error reading Lien: %s", err)
+	}
+	if err := d.Set("restrictions", flattenResourceManagerLienRestrictions(res["restrictions"])); err != nil {
+		return fmt.Errorf("Error reading Lien: %s", err)
+	}
+
+	return nil
+}
+
+func resourceResourceManagerLienDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	url, err := replaceVars(d, config, "https://cloudresourcemanager.googleapis.com/v1/liens?parent={{parent}}")
+	if err != nil {
+		return err
+	}
+
+	url, err = replaceVars(d, config, "https://cloudresourcemanager.googleapis.com/v1/{{name}}")
```

Comment:
This doesn't sound right. Unless name includes `lien/{project_number}`

Reply:
It does, unfortunately - the result back from the server is `lien/a-long-uuid`.  Nothing for it.  :(